test/index.js

```diff
@@ -157,13 +157,49 @@ describe('hpal', () => {
 
                         expect(result.err).to.be.instanceof(DisplayError);
                         expect(result.output).to.equal('');
-                        expect(result.errorOutput).to.contain('It\'s ambiguous which directory containing a .hc.js file to use: ');
-                        expect(result.errorOutput).to.contain('project-a');
-                        expect(result.errorOutput).to.contain('project-b');
+                        expect(result.errorOutput).to.contain('It\'s ambiguous which directory containing a .hc.js file to use: project-a/.hc.js, project-b/.hc.js');
                         expect(result.errorOutput.match(/,/g)).to.have.length(1);
                     });
             });
 
+            it('succeeds when finding a .hc.js file is ambiguous in the project, but not to the cwd (multi-plugin project).', () => {
+
+                return RunUtil.cli(['make', 'route'], 'ambiguous-hc-file/project-a')
+                    .then((result) => {
+
+                        expect(result.err).to.not.exist();
+                        expect(result.output).to.contain('Wrote routes/index.js');
+                        expect(result.errorOutput).to.equal('');
+
+                        return read('ambiguous-hc-file/project-a/routes/index.js');
+                    })
+                    .then((contents) => {
+
+                        expect(contents).to.startWith('\'use strict\';');
+
+                        return rimraf('ambiguous-hc-file/project-a/routes');
+                    });
+            });
+
+            it('succeeds when finding a .hc.js file from a cwd deep in the project.', () => {
+
+                return RunUtil.cli(['make', 'route'], 'non-ambiguous-hc-file-cwd/project-a')
+                    .then((result) => {
+
+                        expect(result.err).to.not.exist();
+                        expect(result.output).to.contain('Wrote ../project-b/routes/index.js');
+                        expect(result.errorOutput).to.equal('');
+
+                        return read('non-ambiguous-hc-file-cwd/project-b/routes/index.js');
+                    })
+                    .then((contents) => {
+
+                        expect(contents).to.startWith('\'use strict\';');
```

Comment:
couple of things:

1. the `read` helper broke my brain a little bit. so ill!
2. this caught my eye; why assert against "use strict"?  I think curious b/c this read as not specific to the file created, so I guess not descriptive (to human reader me :) ) of what success looks like, but also figure that's a super surface-level thought

Reply:
That's a good point.  It is a really superficial check, and I can see the possibility for false positives as you pointed out.  The thing is, hpal is only responsible for a small amount of the contents of this file, and the rest comes from haute-couture, so it's pretty hard to fingerprint these files.  I will think about it though! Thanks for the thought 🙏

Maybe we could ensure the file does not exist before running the command :)

Reply:
oh very cool, yea, didn't think about the existence check at all
👍 mm, all sounds good. thanks for that